base/range.jl

```diff
@@ -500,7 +500,8 @@ julia> step(range(2.5, stop=10.9, length=85))
 """
 step(r::StepRange) = r.step
 step(r::AbstractUnitRange{T}) where{T} = oneunit(T) - zero(T)
-step(r::StepRangeLen{T}) where {T} = T(r.step)
+step(r::StepRangeLen) = r.step
+step(r::StepRangeLen{T}) where {T<:AbstractFloat} = T(r.step)
```

Comment:
What happens if you're even more selective here and restrict it to `StepRangeLen{T,TwicePrecision{T},TwicePrecision{T}}`? Any problems?